src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const stateHistory = [];
+
+  actions.forEach((action) => {
+    let nextState = { ...currentState };
```

Comment:
[NAMING] - The name `nextState` does not accurately represent what this variable is used for. It's better to rename it to `stateCopy` or `currentStateCopy` to reflect that it's a copy of the `currentState` object.